Sources/Overlays/_Testing_Foundation/Attachments/Attachment+URL.swift

```diff
@@ -83,79 +77,134 @@ extension Attachment where AttachableValue == Data {
       // Load the file.
-      try self.init(Data(contentsOf: url, options: [.mappedIfSafe]), named: preferredName, sourceLocation: sourceLocation)
+      try Data(contentsOf: url, options: [.mappedIfSafe])
     }
+#endif
+
+    self.init(data, named: preferredName, sourceLocation: sourceLocation)
   }
 }
 
-// MARK: - Attaching directories
+/// Compress the contents of a directory to an archive, then map that archive
+/// back into memory.
+///
+/// - Parameters:
+///   - directoryURL: A URL referring to the directory to attach.
+///
+/// - Returns: An instance of `Data` containing the compressed contents of the
+///   given directory.
+///
+/// - Throws: Any error encountered trying to compress the directory, or if
+///   directories cannot be compressed on this platform.
+///
+/// This function asynchronously compresses the contents of `directoryURL` into
+/// an archive (currently of `.zip` format, although this is subject to change.)
+private func compressContentsOfDirectory(at directoryURL: URL) async throws -> Data {
+#if !SWT_NO_PROCESS_SPAWNING
+  let temporaryName = "\(UUID().uuidString).zip"
+  let temporaryURL = FileManager.default.temporaryDirectory.appendingPathComponent(temporaryName)
+  defer {
+    try? FileManager().removeItem(at: temporaryURL)
+  }
 
-extension Data {
-  /// Initialize an instance of this type by compressing the contents of a
-  /// directory.
-  ///
-  /// - Parameters:
-  ///   - directoryURL: A URL referring to the directory to attach.
-  ///
-  /// - Throws: Any error encountered trying to compress the directory, or if
-  ///   directories cannot be compressed on this platform.
-  ///
-  /// This initializer asynchronously compresses the contents of `directoryURL`
-  /// into an archive (currently of `.tgz` format, although this is subject to
-  /// change) and stores a mapped copy of that archive.
-  init(compressedContentsOfDirectoryAt directoryURL: URL) async throws {
-    let temporaryName = "\(UUID().uuidString).tgz"
-    let temporaryURL = FileManager.default.temporaryDirectory.appendingPathComponent(temporaryName)
+  try await withCheckedThrowingContinuation { continuation in
+    let process = Process()
 
-#if !SWT_NO_PROCESS_SPAWNING
-#if os(Windows)
-    let tarPath = #"C:\Windows\System32\tar.exe"#
+    // The standard version of tar(1) does not (appear to) support writing PKZIP
+    // archives. FreeBSD's (AKA bsdtar) was long ago rebased atop libarchive and
+    // knows how to write PKZIP archives, while Windows inherited FreeBSD's tar
+    // tool in Windows 10 Build 17063 (per https://techcommunity.microsoft.com/blog/containers/tar-and-curl-come-to-windows/382409).
+    //
+    // On Linux (which does not have FreeBSD's version of tar(1)), we can use
+    // zip(1) instead.
+#if os(Linux)
+    let archiverPath = "/usr/bin/zip"
+#elseif SWT_TARGET_OS_APPLE || os(FreeBSD)
+    let archiverPath = "/usr/bin/tar"
+#elseif os(Windows)
+    let archiverPath = #"C:\Windows\System32\tar.exe"#
```

Comment:
We really should not hardcode this to `C:`. This is a common way to break if Windows is installed to an alternate location. `%SystemRoot%\system32\tar.exe` should be preferred. Even better would be to use `GetSystemDirectoryW` to get the location.

Reply:
I see no reason I can't use `GetSystemDirectoryW` here.